src/mongoose_internal_databases.erl

```diff
@@ -28,11 +30,32 @@ init(mnesia, #{}) ->
             ok
     end,
     application:start(mnesia, permanent),
-    mnesia:wait_for_tables(mnesia:system_info(local_tables), infinity),
+    wait_for_tables_loop(mnesia:system_info(local_tables), 10000, 0),
     mongoose_node_num_mnesia:init();
 init(cets, #{}) ->
     ok.
 
+%% Sometimes mnesia:wait_for_tables/2 could hang on startup.
+%% This function logs which tables are not ready and their status.
+wait_for_tables_loop(Tables, Interval, Total) ->
```

Comment:
This seems unrelated. If there is a new issue, it should be logged and resolved separately IMO.